### ISSUE TITLE
CVXPY correctly detecting CPLEX but asking for a MIP solver

### ISSUE THREAD

**Issue by filippobistaffa** [cvxgrp/cvxpy#1228]
I am trying to solve a MIP with CPLEX (which is a MIP-capable solver according to the [documentation](https://www.cvxpy.org/tutorial/advanced/index.html#mixed-integer-programs)), but even though CVXPY correctly detects the solver (`slv_def.INSTALLED_MI_SOLVERS` is correctly equal to `['CPLEX']`) I still receive the following message:
```
You need a mixed-integer solver for this model. Refer to the documentation
    https://www.cvxpy.org/tutorial/advanced/index.html#mixed-integer-programs
for discussion on this topic.
```
Am I missing something? In particular, the source of my confusion is the following line of code:
https://github.com/cvxgrp/cvxpy/blob/8ac508a3214a0f7045253e27cf41189a8c9062dc/cvxpy/problems/problem.py#L611
In my case, `len(slv_def.INSTALLED_MI_SOLVERS) == 1` because it's a list containing only `'CPLEX'` and `solver != s.ECOS_BB` because it's CPLEX, but why should the error be thrown in this case?

**Comment by rileyjmurray** [comment]
The check for a single solver was based on the assumption that ECOS_BB would always be installed (since it comes with ECOS and ECOS is a cvxpy dependency). The logic of checking ``solver != s.ECOS_BB`` is that although ECOS_BB is always available, the user must explicitly call ECOS_BB in order to use it. The issue here is that ``slv_def.INSTALLED_MI_SOLVERS`` doesn't include ECOS_BB in your case. Can you provide us with more information about your system? Where did you get your cvxpy distribution?

**Comment by filippobistaffa** [comment]
I am using Ubuntu 18.04.5 and Python 3.6.9. I installed CVXPY with `sudo /usr/bin/python3.6 -m pip install cvxpy`. Probably the missing dependency comes from the fact that I previously installed CVXPY for Python 3.8 (which I also have installed on this system), and later I removed and re-installed it for Python 3.6.9. Maybe something went wrong along the way. So I should simply re-install the missing dependency? How do I manually do that?
